core/src/main/java/de/eldoria/bloodnight/mob/CustomMob.java

```diff
@@ -6,13 +6,46 @@
 import de.eldoria.bloodnight.mob.meta.Extension;
 import de.eldoria.bloodnight.nodes.container.ContainerMeta;
 import de.eldoria.bloodnight.nodes.container.NodeContainer;
+import de.eldoria.bloodnight.util.Checks;
+import de.eldoria.bloodnight.util.MobTags;
 import org.bukkit.entity.Entity;
+import org.bukkit.entity.Mob;
+import org.bukkit.persistence.PersistentDataType;
 
 public record CustomMob(String id, Equipment equipment, Attributes attributes, NodeContainer nodes, MobDrops mobDrops,
                         Extension extension) {
-    public CustomMob copy(Entity entity, Entity extension) {
+
+    public CustomMob {
+        Checks.notNull(id, "Mob id can not be null");
+        Checks.lowerCase(id, "Mob id must be lower case.");
+    }
+
+    public CustomMob createLiving(Entity entity) {
+        MobTags.markCustomMob(entity);
+
         NodeContainer copy = nodes.copy();
-        copy.inject(new ContainerMeta(entity, extension));
+        Entity extensionEntity = createExtension(entity);
+
+        // TODO: 21.08.23 apply other attributes
+
+        copy.inject(new ContainerMeta(entity, extensionEntity));
         return new CustomMob(id, equipment, attributes, copy, mobDrops, this.extension);
     }
+
+    private Entity createExtension(Entity baseEntity) {
+        if (extension == null) return null;
```

Comment:
Extension should never be null, but its ExtensionType should be set to NONE.
This is to preserve data already set when a mob is copied and to make it easier for the frontend and not make it build up default data.